Lobby/EOSLobby.cs

```diff
@@ -0,0 +1,197 @@
+
+    private string currentLobbyId = string.Empty;
+    private bool isLobbyOwner = false;
+
+    //events
+    public delegate void JoinLobbyHandler(List<Attribute> attributes);
+    public event JoinLobbyHandler JoinLobbyComplete;
+
+    public delegate void FindLobbiesHandler(List<LobbyDetails> foundLobbies);
+    public event FindLobbiesHandler FindLobbiesComplete;
+
+    //creates a lobby
+    public virtual void CreateLobby(uint maxConnections, LobbyPermissionLevel permissionLevel, bool presenceEnabled, AttributeData[] lobbyData = null)
+    {
+        EOSSDKComponent.GetLobbyInterface().CreateLobby(new CreateLobbyOptions 
+        { 
+            //lobby options
+            LocalUserId = EOSSDKComponent.LocalUserProductId, 
+            MaxLobbyMembers = maxConnections,
+            PermissionLevel = permissionLevel,
+            PresenceEnabled = presenceEnabled,
+        }, null, (CreateLobbyCallbackInfo callback) => 
+        {
+            List<Attribute> lobbyReturnData = new List<Attribute>();
+
+            //if the result of CreateLobby is not successful, log and error and return
+            if(callback.ResultCode != Result.Success)
+            {
+                Debug.LogError("There was an error while creating a lobby. Error: " + callback.ResultCode);
+                return;
+            }
+
+            //create mod handle and lobby data
+            LobbyModification modHandle = new LobbyModification();
+            AttributeData defaultData = new AttributeData { Key = DefaultAttributeKey, Value = DefaultAttributeKey };
+            AttributeData hostAddressData = new AttributeData { Key = hostAddressKey, Value = EOSSDKComponent.LocalUserProductIdString };
+
+            //set the mod handle
+            EOSSDKComponent.GetLobbyInterface().UpdateLobbyModification(new UpdateLobbyModificationOptions 
+            { LobbyId = callback.LobbyId, LocalUserId = EOSSDKComponent.LocalUserProductId }, out modHandle);
+
+            //add attributes
+            modHandle.AddAttribute(new LobbyModificationAddAttributeOptions { Attribute = defaultData, Visibility = LobbyAttributeVisibility.Public });
+            modHandle.AddAttribute(new LobbyModificationAddAttributeOptions { Attribute = hostAddressData, Visibility = LobbyAttributeVisibility.Public });
+
+            //add user attributes
+            if (lobbyData != null)
+            {
+                foreach (AttributeData data in lobbyData)
+                {
+                    modHandle.AddAttribute(new LobbyModificationAddAttributeOptions { Attribute = data, Visibility = LobbyAttributeVisibility.Public });
+                    lobbyReturnData.Add(new Attribute { Data = data, Visibility = LobbyAttributeVisibility.Public });
+                }
+            }
+
+            //update the lobby
+            EOSSDKComponent.GetLobbyInterface().UpdateLobby(new UpdateLobbyOptions { LobbyModificationHandle = modHandle }, null, (UpdateLobbyCallbackInfo _) => { });
```

Comment:
The code following this call should be in the callback of this one as only then we should consider the lobby created